src/applications/form-renderer/utils/formConfig.js

```diff
@@ -0,0 +1,78 @@
+import {
+  dateOfBirthSchema,
+  dateOfBirthUI,
+  fullNameSchema,
+  fullNameUI,
+  titleUI,
+} from 'platform/forms-system/src/js/web-component-patterns';
+import manifest from '../manifest.json';
+import IntroductionPage from '../containers/IntroductionPage';
+import ConfirmationPage from '../containers/ConfirmationPage';
+
+const selectSchemas = ({ pageTitle, additionalFields }) => {
+  const schemas = {
+    schema: {
+      type: 'object',
+      properties: {
+        fullName: fullNameSchema,
+      },
+    },
+    uiSchema: {
+      ...titleUI(pageTitle),
+      fullName: fullNameUI(),
+    },
+  };
+
+  if (additionalFields.includeDateOfBirth) {
+    schemas.schema.properties.dateOfBirth = dateOfBirthSchema;
+    schemas.uiSchema.dateOfBirth = dateOfBirthUI();
+  }
+
+  return schemas;
+};
+
+const formatChapters = chapters => {
+  const formattedChapters = {};
+
+  chapters.forEach(chapter => {
+    const pages = {};
+    pages[chapter.id] = {
```

Comment:
**CONSIDER**
It feels like it would be a bit more "immutable" to simply define `pages` to be an object with the key `chapter.id' all in one definition/declaration:

```
const pages = {
  [chapter.id]: {
    ...
  }
}
```

This is a very small issue in this case, but something I personally like to do when possible.

Separately, this one-page-per-chapter code is feeling increasingly odd to me now that we're getting to the point of seeing things in the context of the `formConfig`. It might be the case that we settle on this constraint (one page per chapter - though I'm not sure that's likely), but it feels odd to see us indexing pages with the chapter id. Ultimately, that's not a problem in and of itself, but it feels slightly confusing to me at first glance. When browsing the code, it feels like it _could_ be a mistake. We know it's not, so, for now, perhaps it could be useful to include a comment explaining this one-page-per-chapter setup under which we currently are operating.

Reply:
I did it the way I did solely because I forgot how to define a dynamic key in an object literal. The code has now been updated.

Reply:
> Separately, this one-page-per-chapter code is feeling increasingly odd to me now that we're getting to the point of seeing things in the context of the formConfig. It might be the case that we settle on this constraint (one page per chapter - though I'm not sure that's likely), but it feels odd to see us indexing pages with the chapter id. Ultimately, that's not a problem in and of itself, but it feels slightly confusing to me at first glance. When browsing the code, it feels like it could be a mistake. We know it's not, so, for now, perhaps it could be useful to include a comment explaining this one-page-per-chapter setup under which we currently are operating.

I agree the code looks a little misleading here with `chapter.id` being used as a page key. There's a fundamental disagreement between VA.gov Design and Engineering teams over how VA.gov forms are structured, and we're trying to bridge the divide as best we can. As I mentioned above, if we can get to the point where we can guarantee the uniqueness of chapter and/or page titles in Drupal, we'll have a lot more flexibility to play around with this structure.

In the meantime, I've added a comment explaining why `chapter.id` is appearing as a key within a `pages` object.